include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -1197,10 +1235,9 @@ class concurrent_skip_list {
 
     node_allocator_type my_node_allocator;
     key_compare my_compare;
-    random_level_generator_type my_rng;
     atomic_node_ptr my_head_ptr;
-    std::atomic<size_type> my_size;
```

Comment:
There is also an option to keep `my_size`, reserve a flag bit in it and use it for optimizing consecutive calls to `size()` by saving the result of combine into `my_size` and marking it as "changed" in the insertion.
I did not investigate potential performance effects from CAS operations on my_size, just wanted to save the idea.